apps/resize/Makefile

```diff
@@ -14,55 +14,62 @@ lanczos_float32_up lanczos_float32_down \
 lanczos_uint16_up lanczos_uint16_down \
 lanczos_uint8_up lanczos_uint8_down
 
-LIBRARIES = $(foreach V,$(VARIANTS),$(BIN)/resize_$(V).a)
-OUTPUTS = $(foreach V,$(VARIANTS),$(BIN)/out_$(V).png)
+LIBRARIES = $(foreach V,$(VARIANTS),$(BIN)/%/resize_$(V).a)
+OUTPUTS = $(foreach V,$(VARIANTS),$(BIN)/$(HL_TARGET)/out_$(V).png)
 
 all: $(OUTPUTS)
 
-$(BIN)/resize.generator: resize_generator.cpp $(GENERATOR_DEPS)
+$(GENERATOR_BIN)/resize.generator: resize_generator.cpp $(GENERATOR_DEPS)
 	@mkdir -p $(@D)
 	$(CXX) $(CXXFLAGS) -g $(filter-out %.h,$^) -o $@ $(LDFLAGS) $(HALIDE_SYSTEM_LIBS)
 
-$(BIN)/resize_%.a: $(BIN)/resize.generator
-	@mkdir -p $(@D)
-	$^ -g resize -o $(BIN) -f resize_$* \
-	target=$(HL_TARGET)-no_runtime \
-	interpolation_type=$$(echo $* | cut -d_ -f1) \
-	input.type=$$(echo $* | cut -d_ -f2) \
-	upsample=$$(echo $* | cut -d_ -f3 | sed 's/up/true/;s/down/false/')
 
-$(BIN)/runtime.a: $(BIN)/resize.generator
+# Can't have multiple wildcards in Make, so we'll use a macro
```

Comment:
Yuck, but good to know how to do this.

Reply:
...in Bazel this would just be a list comprehension :-)

Reply:
I think this is my single favorite feature of bazel/blaze.